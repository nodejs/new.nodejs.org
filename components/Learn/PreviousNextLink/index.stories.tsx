import PrevNextLink from '.';
import type { Meta as MetaObj, StoryObj } from '@storybook/react';

type Story = StoryObj<typeof PrevNextLink>;
type Meta = MetaObj<typeof PrevNextLink>;

export const Default: Story = {
  args: {
    previous: { slug: '/previous' },
    next: { slug: '/next' },
  },
};

export const WithoutNext: Story = {
  args: {
    previous: { slug: '/previous' },
  },
};

export const WithoutPrevious: Story = {
  args: {
    next: { slug: '/next' },
  },
};

export default { component: PrevNextLink } as Meta;
