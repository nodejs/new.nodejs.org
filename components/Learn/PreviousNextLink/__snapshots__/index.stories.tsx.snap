// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Learn/PreviousNextLink Default smoke-test 1`] = `
<ul class="PreviousNextLink_prevNextLink__9SAYu">
  <li>
    <a rel="prev"
       href="/en/previous"
    >
      <svg stroke="currentColor"
           fill="currentColor"
           stroke-width="0"
           viewbox="0 0 448 512"
           height="1em"
           width="1em"
           xmlns="http://www.w3.org/2000/svg"
           style="margin-right: 5px;"
      >
        <path d="M223.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L319.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L393.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34zm-192 34l136 136c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9L127.9 256l96.4-96.4c9.4-9.4 9.4-24.6 0-33.9L201.7 103c-9.4-9.4-24.6-9.4-33.9 0l-136 136c-9.5 9.4-9.5 24.6-.1 34z">
        </path>
      </svg>
      PREV
    </a>
  </li>
  <li>
    <a rel="next"
       href="/en/next"
    >
      NEXT
      <svg stroke="currentColor"
           fill="currentColor"
           stroke-width="0"
           viewbox="0 0 448 512"
           height="1em"
           width="1em"
           xmlns="http://www.w3.org/2000/svg"
           style="margin-left: 5px;"
      >
        <path d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34zm192-34l-136-136c-9.4-9.4-24.6-9.4-33.9 0l-22.6 22.6c-9.4 9.4-9.4 24.6 0 33.9l96.4 96.4-96.4 96.4c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l136-136c9.4-9.2 9.4-24.4 0-33.8z">
        </path>
      </svg>
    </a>
  </li>
</ul>
`;

exports[`Learn/PreviousNextLink WithoutNext smoke-test 1`] = `
<ul class="PreviousNextLink_prevNextLink__9SAYu">
  <li>
    <a rel="prev"
       href="/en/previous"
    >
      <svg stroke="currentColor"
           fill="currentColor"
           stroke-width="0"
           viewbox="0 0 448 512"
           height="1em"
           width="1em"
           xmlns="http://www.w3.org/2000/svg"
           style="margin-right: 5px;"
      >
        <path d="M223.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L319.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L393.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34zm-192 34l136 136c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9L127.9 256l96.4-96.4c9.4-9.4 9.4-24.6 0-33.9L201.7 103c-9.4-9.4-24.6-9.4-33.9 0l-136 136c-9.5 9.4-9.5 24.6-.1 34z">
        </path>
      </svg>
      PREV
    </a>
  </li>
  <li>
  </li>
</ul>
`;

exports[`Learn/PreviousNextLink WithoutPrevious smoke-test 1`] = `
<ul class="PreviousNextLink_prevNextLink__9SAYu">
  <li>
  </li>
  <li>
    <a rel="next"
       href="/en/next"
    >
      NEXT
      <svg stroke="currentColor"
           fill="currentColor"
           stroke-width="0"
           viewbox="0 0 448 512"
           height="1em"
           width="1em"
           xmlns="http://www.w3.org/2000/svg"
           style="margin-left: 5px;"
      >
        <path d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34zm192-34l-136-136c-9.4-9.4-24.6-9.4-33.9 0l-22.6 22.6c-9.4 9.4-9.4 24.6 0 33.9l96.4 96.4-96.4 96.4c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l136-136c9.4-9.2 9.4-24.4 0-33.8z">
        </path>
      </svg>
    </a>
  </li>
</ul>
`;
